docs/developer/sdk_from_swagger_and_typespec.md

```diff
@@ -0,0 +1,195 @@
+# Overview
+
+This document compares Python SDKs generated from Swagger (OpenAPI) specifications versus TypeSpec. For clarity, we'll refer to these as "Swagger SDKs" and "TypeSpec SDKs" respectively.
+
+## Model 
+
+### Model Structure
+
+#### Msrest Model
+
+Swagger SDKs are generated from [Swagger specifications](https://github.com/Azure/azure-rest-api-specs/tree/main/specification) using [@autorest/python](https://www.npmjs.com/package/@autorest/python), and implement the Msrest model pattern. The following example illustrates the fundamental structure of an Msrest model:
+
+```python
+from azure.mgmt.example._utils import serialization as _serialization
+
+class Person(_serialization.Model):
+    _attribute_map = {
+        "name": {"key": "name", "type": "str"},
+        "parent_name": {"key": "parentName", "type": "str"},
+    }
+
+    def __init__(self, *, name: Optional[str] = None, parent_name: Optional[str] = None) -> None:
+        ...
+```
+
+### DPG Model
+
+TypeSpec SDKs are generated from [TypeSpec](https://github.com/microsoft/typespec/) using [@azure-tools/typespec-python](https://www.npmjs.com/package/@azure-tools/typespec-python), and implement the DPG model pattern. The following example demonstrates the fundamental structure of a DPG model:
+
+```python
+from azure.mgmt.example._utils.model_base import Model as _Model, rest_field
+
+class Person(_Model):
+    name: Optional[str] = rest_field()
+    parent_name: Optional[str] = rest_field(name="parentName")
+
+    @overload
+    def __init__(
+        self,
+        *,
+        name: Optional[str] = None,
+        parent_name: Optional[str] = None,
+    ) -> None: ...
+
+    @overload
+    def __init__(self, mapping: Mapping[str, Any]) -> None: ...
+
+
+    def __init__(self, *args: Any, **kwargs: Any) -> None:
+        super().__init__(*args, **kwargs)
+```
+
+### Model Usage
+
+#### Msrest Model Usage
+
+```python
+msrest_model = Person(name="xxx", parent_name="xxx")
+print(msrest_model.name)
+print(msrest_model.parent_name)
+
+# Access model as a dictionary
+json_model = msrest_model.as_dict()
+print(json_model["name"])
+print(json_model["parentName"])
+```
+
+#### DPG Model Usage
+
+```python
+dpg_model = Person(name="xxx", parent_name="xxx")
+print(dpg_model.name)
+print(dpg_model.parent_name)
+
+# Access model directly as a dictionary
+print(dpg_model["name"])
+print(dpg_model["parentName"])
+```
+
+By comparing these usage patterns, we can see that DPG models can be accessed directly as dictionaries without calling `.as_dict()`, providing a more convenient experience.
+
+#### Usage Note
+
+For backward compatibility, DPG models continue to support the `.as_dict()` method for existing SDK users.
+
+### Model Flattening
```

Comment:
I think we can jsut stick with simple, and have a note about how nested flattening etc is supported in msrest but not typespec. No need to add examples for these more complex scenarios, i think we can just briefly talk about iot

Reply:
Since it is doc for internal use, I think it is easy to describe and understand if there is an example.